atuin-client/src/history.rs

```diff
@@ -62,10 +98,241 @@ impl History {
             session,
             hostname,
             deleted_at,
+            interpreter,
+            context,
+            _seal: HistorySeal,
         }
     }
 
+    /// Builder for a history entry that is imported from shell history.
```

Comment:
here and below are new builders, with some documentation considering their scenarios and usage